packages/editor/src/components/blocks/datatable/DataTable.tsx

```diff
@@ -0,0 +1,92 @@
+import { createInitiTableColumns } from '../../../data/data';
+import { IvyIcons } from '@axonivy/ui-icons';
+
+type DataTableProps = Prettify<DataTable>;
+
+export const defaultDataTableProps: DataTable = {
+  components: [],
+  label: 'Label',
+  value: '',
+  ...defaultBaseComponent
+} as const;
+
+export const DataTableComponent: ComponentConfig<DataTableProps> = {
+  name: 'DataTable',
+  category: 'Elements',
+  subcategory: 'Input',
+  icon: <IconSvg />,
+  description: 'A datatable component',
+  defaultProps: defaultDataTableProps,
+  render: props => <UiBlock {...props} />,
+  create: ({ label, value, ...defaultProps }) => ({ ...defaultDataTableProps, label, value, ...defaultProps }),
+  outlineInfo: component => component.label,
+  fields: {
+    components: { subsection: 'General', type: 'hidden' },
+    label: { subsection: 'General', label: 'Label', type: 'text' },
+    value: { subsection: 'General', label: 'Value', type: 'textBrowser', options: { onlyTypesOf: 'List<' } },
+    ...baseComponentFields
+  }
+};
+
+const UiBlock = ({ id, label, components, value }: UiComponentProps<DataTableProps>) => (
+  <div className='block-table'>
+    <div className='block-table__label'>
+      <span>{label}</span>
+      <span style={{ color: 'var(--N600)' }}>{value.length > 0 ? value : 'value is empty'}</span>
+    </div>
+    {components.length > 0 && (
+      <div className='block-table__columns'>
+        {components.map((column, index) => (
+          <ComponentBlock key={column.id} component={column} preId={components[index - 1]?.id} />
+        ))}
+      </div>
+    )}
+    {components.length === 0 && <DataTableColumnDialog id={id} initValue={value} />}
+  </div>
+);
+
+const DataTableColumnDialog = ({ id, initValue }: { id: string; initValue: string }) => {
+  const { context, setData } = useAppContext();
+  const [tree, setTree] = useState<Array<BrowserNode<Variable>>>([]);
+  const dataClass = useMeta('meta/data/attributes', context, { types: {}, variables: [] }).data;
+  useEffect(() => setTree(findAttributesOfType(dataClass, initValue)), [dataClass, initValue]);
+
+  const createColumns = () => {
+    setData(data => {
+      const creates = tree[0].children
+        .map(attribute => {
+          const component = componentByName('DataTableColumn');
+          if (component === undefined) {
+            return undefined;
+          }
+          return {
+            componentName: component.name,
+            label: attribute.value,
+            value: attribute.value
+          };
+        })
+        .filter(create => create !== undefined);
+      return createInitiTableColumns(id, data, creates);
+    });
+  };
+
+  return initValue.length > 0 ? (
+    <Button icon={IvyIcons.DatabaseLink} variant='outline' onClick={createColumns}>
+      Create columns from value
+    </Button>
+  ) : (
+    <Message variant='warning' message='Value of DataTable is empty. Define value within Properties of DataTable' />
+  );
```

Comment:
instead of this, I would fast return, e.g. like:
```tsx
if (initValue.length === 0) {
  return <Message ... />
}

const createColumns = () => {
...
}

return <Button ... />

```